ClassLibrary/ClassLibrary/RepositoryBooks.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace ClassLibrary
+{
+    public class RepositoryBooks
+    {
+        // looks like with "DepartmentReaders", you can write for one occasion...  
+
+        //for comfortable I do it open, but I break AV1014 (An object should not expose any other classes it depends on for Library)
+        private List<Book> Books
+        {
+            get;
+            set;
+        }
+            = new List<Book>();
+
+        public void AddBook(Book book)
+        {
+            if (book == null)
+            {
+                throw new ArgumentNullException("Book's reference is null");
+            }
+
+            Book foundBook = Books?.FirstOrDefault(x => x.ISBN == book.ISBN);
+            if (foundBook != null)
+            {
+                throw new ArgumentException($"This book's [{book.ISBN},{book.NameBook}] exists");
+            }
+            else
+            {
+                Books.Add(book);
+            }
+        }
+
+        // what should we return ?
+        // what AV1130? I return "list" (For convenience), but use IEnumerable inside 
+        // And how should I convert the list (just below)? 
+        public List<Book> FindBooks(String NameBook="",String NameAutor="",String LastNameAutor="")
+        {
+            IEnumerable < Book > request = null;
+            if (NameBook.IsNullOrWhiteSpace() &&
+                NameAutor.IsNullOrWhiteSpace() &&
+                LastNameAutor.IsNullOrWhiteSpace())
+            {
+                return Books;//??? convert IEnumerable->list?
+            }
+            ;
+            if (!NameBook.IsNullOrWhiteSpace())
+            {
+                request= Books.Where(book => book.NameBook == NameBook).ToList();
+            }
+
+            if (!NameAutor.IsNullOrWhiteSpace())
+            {
+                request =(request == null)?
```

Comment:
there's no need in any type of conditions inside this `if` statement. please do some thinking and make this whole method much simpler.

Reply:
I replace, but I am not sure that better solve (the condition exists)